test/unit/math/test_ad.hpp

```diff
@@ -509,18 +505,14 @@ void expect_ad_vv(const ad_tolerances& tols, const F& f, int x1, const T2& x2) {
 
 template <typename F, typename T1>
 void expect_ad_vv(const ad_tolerances& tols, const F& f, const T1& x1, int x2) {
+  double x2_dbl = static_cast<double>(x2);
+
+  // test value, exception handling checked recursively
   try {
-    f(x1, x2);
+    expect_near_rel("expect_ad_vv(T1, int)", f(x1, x2), f(x1, x2_dbl));
   } catch (...) {
-    expect_all_throw(f, x1, x2);
```

Comment:
@peterwicksstringfield, thanks for adding this code. For the most part it looks ok, but I'm wondering if this is missing a test case.

It looks like with this change, if `f(x1, x2)` or `f(x1, x2_dbl)` throws, but not both, then we miss marking that as an error. (The old code had `expect_all_throw(f, x1, x2)`.)

I saw that there were some tests below, so maybe this has the right behavior and I missed it. So... question: will this fail a test if either one throws but not the other?

Reply:
It is missing a testcase, just as you say. It will NOT fail a test if one throws and the other does not. The negated version of expect_ad in this PR is broken too.

Thank you for taking the time to review and for spotting this.